source-code/component/menu/menu.html

```diff
@@ -0,0 +1,314 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+    <title>menu</title>
+    <style>
```

Comment:
the file is a little hard to follow, consider extracting the CSS in an external file